src/main/java/org/codehaus/mojo/jaxws/AbstractJaxwsMojo.java

```diff
@@ -435,7 +435,7 @@ protected void exec( List<String> arguments )
             {
                 for ( String arg : vmArgs )
                 {
-                    cmd.createArg().setLine( arg );
+                    cmd.createArg().setValue( arg );
                 }
```

Comment:
The call to setLine removes all single and double quotes. To get around this I've used setValue for this

Reply:
This is a potentially breacking change. Can you please post an example value that would use in here pls?

Reply:
I thought about that. There where several reasons why I will take a chance on the risk:

1. The system properties where introduced only a while ago and are (currently?) not part of the original MojoHaus plugin. So they look kinda fix, enhancement or tryout to me right now.
1. The only VmArg used in the plugin right now is the http.nonProxyHosts. So there are no side effects other code.
1. With the settings in our build environment the plugin breaks even on Java 8 compared to the MojoHaus plugin. In my point of view it must be either fixed or reversed to the original code base.

Thinking further I guess that the replacement of "|" with "^|" is a solution that is unique to the cmd escaping mechanism of windows. On Linux or other operating systems this (might) not work. To improve this we either have to do the escaping individually depending on the underlying OS.

I also guess that the plugin needs additional tests on at least the windows and linux platform to prove that the implementation is building the command line with correct escaping.

In the end I'm an agile guy. Better try out and get response, then improve or roll forward.

Said so, it's up to you to decide about taking the risk coming with this pull request.